documentation/api/trace-custom.md

```diff
@@ -0,0 +1,126 @@
+# Trace - Get Custom
+
+Captures a diagnostic trace of a process using the given set of event providers specified in the request body.
+
+## HTTP Route
+
+```http
+POST https://localhost:52323/trace/{pid}?durationSeconds={durationSeconds}
+```
+
+or 
+
+```http
+POST https://localhost:52323/trace/{uid}?durationSeconds={durationSeconds}
+```
+
+or
+
+```http
+POST https://localhost:52323/trace?durationSeconds={durationSeconds}
+```
+
+> **NOTE:** Process information (IDs, names, environment, etc) may change between invocations of these APIs. Processes may start or stop between API invocations, causing this information to change.
+
+## URI Parameters
+
+| Name | In | Required | Type | Description |
+|---|---|---|---|---|
+| `pid` | path | false | int | The ID of the process. |
+| `uid` | path | false | guid | A value that uniquely identifies a runtime instance within a process. |
+| `durationSeconds` | query | false | int | The duration of the trace operation in seconds. Default is `30`. Min is `-1` (indefinite duration). Max is `2147483647`. |
+
+See [ProcessIdentifier](definitions.md#ProcessIdentifier) for more details about these parameters.
+
+If neither `pid` nor `uid` are specified, a trace of the [default process](defaultprocess.md) will be captured. Attempting to capture a trace of the default process when the default process cannot be resolved will fail.
+
+## Authentication
+
+Allowed schemes:
+- `MonitorApiKey`
+- `Negotiate` (Windows only, running as unelevated)
+
+See [Authentication](./../authentication.md) for further information.
+
+## Request Body
+
+A request body of type [EventProvidersConfiguration](definitions.md#EventProvidersConfiguration) is required.
+
+The expected content type is `application/json`.
+
+## Responses
+
+| Name | Type | Description | Content Type |
+|---|---|---|---|
+| 200 OK | stream | A trace of the process. | `application/octet-stream` |
+| 400 Bad Request | [ValidationProblemDetails](definitions.md#ValidationProblemDetails) | An error occurred due to invalid input. The response body describes the specific problem(s). | `application/problem+json` |
+| 401 Unauthorized | | Authentication is required to complete the request. See [Authentication](./../authentication.md) for further information. | |
+| 429 Too Many Requests | | There are too many trace requests at this time. Try to request a trace at a later time. | |
+
+> **NOTE:** After the expiration of the trace duration, completing the request may take a long time (up to several minutes) for large applications if `EventProvidersConfiguration.RequestRundown` is set to `true`. The runtime needs to send over the type cache for all managed code that was captured in the trace, known as rundown events. Thus, the length of time of the request may take significantly longer than the requested duration.
+
+## Examples
+
+### Sample Request
+
+```http
+POST https://localhost:52323/trace/21632?durationSeconds=60
```

Comment:
```suggestion
POST /trace/21632?durationSeconds=60 HTTP/1.1
Host: localhost:52323/
```

I thought specifying the FQDN in the first line was against the HTTP spec? Or at least an esoteric way to do it

Reply:
Sure, I can go through everything and reformat it this way.

Reply:
I'll fix up the dump ones in the other PR.